### PR TITLE
Signing cleanup

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Now that #38457 is merged in, there's more cleanup we can do of the signing infrastructure in the repo.

This PR moves the CrossOS DAC signing to be done in the CrossDac Pack job. As a result, we can remove the signing at build time and remove the signing during prepare-artifacts.

I also renamed UnsignedArtifacts->IntermediateArtifacts because some of the artifacts are signed and the final artifacts after prepare-artifacts are still unsigned now that we only support post build signing.

Official build: https://dev.azure.com/dnceng/internal/_build/results?buildId=885893&view=results

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
Looks like I've got a little more work to do relating to the hosting packages (they still use the old tooling) before I can do this.

File: src/installer/pkg/projects/host-packages.proj [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<Project>
+  <Import Sdk="Microsoft.Build.Traversal" Project="Sdk.props" />
+
+  <ItemGroup>
+    <Project Include="Microsoft.NETCore.DotNetHost\Microsoft.NETCore.DotNetHost.pkgproj" />
+    <Project Include="Microsoft.NETCore.DotNetHostPolicy\Microsoft.NETCore.DotNetHostPolicy.pkgproj" />
+    <Project Include="Microsoft.NETCore.DotNetHostResolver\Microsoft.NETCore.DotNetHostResolver.pkgproj" />
+    <ProjectReference Include="@(Project)" />
+    <ProjectReference Include="@(Project)" AdditionalProperties="PackageTargetRuntime=$(PackageRID)" />
```
Comment (ViktorHofer) [inline src/installer/pkg/projects/host-packages.proj:9]:
```suggestion
    <ProjectReference Include="Microsoft.NETCore.DotNetHost\Microsoft.NETCore.DotNetHost.pkgproj" />
    <ProjectReference Include="Microsoft.NETCore.DotNetHostPolicy\Microsoft.NETCore.DotNetHostPolicy.pkgproj" />
    <ProjectReference Include="Microsoft.NETCore.DotNetHostResolver\Microsoft.NETCore.DotNetHostResolver.pkgproj" />
    <ProjectReference Update="@(ProjectReference)" AdditionalProperties="PackageTargetRuntime=$(PackageRID)" />
```

**Review by ViktorHofer — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Updated official build: https://dev.azure.com/dnceng/internal/_build/results?buildId=887787&view=results

**Comment by jkoritzinsky** [comment]
Looks like our build are getting NuGet rate limited again (at least my PR is).

**Comment by ViktorHofer** [comment]
https://github.com/dotnet/runtime/issues/44700

**Review by safern — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
The official build passed. I'll try to get a passing build here as well.

**Comment by jkoritzinsky** [comment]
Errors are the usual errors we've been seeing with the Android runtime tests. Since this PR only affects CoreCLR and installer and the official build passed, I'm going to merge this in.

**Pull request merged**